activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -105,6 +106,18 @@ class ReferencesForeignKeyTest < ActiveRecord::TestCase
           @connection.remove_reference :testings, :testing_parent, foreign_key: true
         end
       end
+
+      test "foreign key column can be added for singular table" do
+        ActiveRecord::Base.stubs(:pluralize_table_names).returns(false)
```

Comment:
instead of stubbing, we usually assign the value and restore the configuration. Something along the lines of:

```ruby
begin
  original_pluralize_table_names = ActiveRecord::Base.pluralize_table_names
  ActiveRecord::Base.pluralize_table_names = false
ensure
  ActiveRecord::Base.pluralize_table_names = original_pluralize_table_names
end
```